sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_intel_matrix.asciidoc

```diff
@@ -21,135 +19,380 @@
-value to determine which of the extension's APIs the implementation supports.
+specification. An implementation supporting this extension must
+predefine the macro `SYCL_EXT_INTEL_MATRIX` to one of the values
+defined in the table below.Applications can test for the existence of
+this macro to determine if the implementation supports this feature,
+or applications can test the macro's value to determine which of the
+extension's APIs the implementation supports.
 
-[frame="none",options="header"]
-|======================
-|Value |Description
-|1     |Introduce `packed` layout and extend `joint_matrix_store` to Matrix A and B.
-|======================
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
 
 
-## Extra Functionality
+=== Joint Matrix Intel-Specific Matrix Features
 
-### Layout
-Besides row major and column major layouts, `layout` introduces the custom layout packed layout that refers to the VNNI format descibed in the following section.
+==== Layout
+Besides row major and column major layouts, `layout` introduces the
+custom layout packed layout that refers to the VNNI format descibed in
+the following section.
 
 ```c++
 namespace sycl::ext::intel::experimental::matrix {
+
 enum class layout {
   packed
 };
-}
+
+} // namespace sycl::ext::intel::experimental::matrix
 ```
 
 
-### Layout argument in `joint_matrix_load`
-`layout` in `joint_matrix_load` can take `packed` as argument to specify that the data has already been transformed into VNNI format (`packed`). in this case, `stride` argument of `joint_matrix_load` describes the number of elements between consecutive rows for packed layouts.
+==== Layout argument in `joint_matrix_load`
+`layout` in `joint_matrix_load` can take `packed` as argument to
+specify that the data has already been transformed into VNNI format
+(`packed`). in this case, `stride` argument of `joint_matrix_load`
+describes the number of elements between consecutive rows for packed
+layouts.
+
+In order to get maximum performance on Intel AMX and Intel XMX,
+prepacking data in the memory is necessary. If users did not specify
+the packed layouts, transforms done by the implementation will be slow
+due to extra scatter/gather operations. Hence, we expose the `packed`
+layout to the user to specify that A or B have already been
+VNNIed. The packed or VNNI layout is introduced in the `VNNI layout`
+section below.
+
+IMPORTANT: In the current Intel AMX and Intel XMX implementations, the
+layout in the load of matrix B (provided by the `layout memL`
+parameter below) must be `packed` or `row_major`. Automatic VNNI
+transform is supported on AMX. The layout in the load of matrices A
+and C must be `row_major`, and the layout in the store of matrix C
+(provided by the `layout memL` parameter below) must also be
+`row_major`.
```

Comment:
Yes this is the note I talk about in my comment above. I will remove it because this is temporary. Support of all layouts on AMX is done. Support for col major on PVC is in progress.